Helper/Entity/ProductHelper.php

```diff
@@ -263,6 +256,28 @@ public function getProductCollectionQuery(
         return $products;
     }
 
+    protected function addStockFilter($products, $storeId)
+    {
+        if ($this->configHelper->getShowOutOfStock($storeId) === false) {
+            $this->stockHelper->addInStockFilterToCollection($products);
+        }
+
+        return $products;
+    }
+
+    protected function addMandatoryAttributes($products)
+    {
+        /** @var \Magento\Catalog\Model\ResourceModel\Product\Collection $products */
+        $products->addFinalPrice()
+            ->addAttributeToSelect('special_price')
+            ->addAttributeToSelect('special_from_date')
+            ->addAttributeToSelect('special_to_date')
+            ->addAttributeToSelect('visibility')
+            ->addAttributeToSelect('status');
+
+        return $products;
```

Comment:
I think you need to return nothing here. Seems that we are mutating `$products`.

Reply:
done